tests/pipelines/test_pipelines_automatic_speech_recognition.py

```diff
@@ -266,6 +262,49 @@ def test_torch_whisper(self):
         output = speech_recognizer([filename], chunk_length_s=5, batch_size=4)
         self.assertEqual(output, [{"text": " A man said to the universe, Sir, I exist."}])
 
+    @require_torch
+    @slow
+    def test_whisper_timestamp_prediction(self):
+        processor = WhisperProcessor.from_pretrained("openai/whisper-tiny")
+        model = WhisperForConditionalGeneration.from_pretrained("openai/whisper-tiny")
+        model.config.forced_decoder_ids = processor.get_decoder_prompt_ids(
+            task="transcribe", language="en", no_timestamps=False
+        )
+        model.config.begin_suppress_tokens += [50363]
+        ds = load_dataset("hf-internal-testing/librispeech_asr_dummy", "clean", split="validation").sort("id")
+        array = np.concatenate(
+            [ds[40]["audio"]["array"], ds[41]["audio"]["array"], ds[42]["audio"]["array"], ds[43]["audio"]["array"]]
+        )
+        pipe = pipeline(
+            "automatic-speech-recognition",
+            model=model,
+            tokenizer=processor.tokenizer,
+            feature_extractor=processor.feature_extractor,
+            forced_bos_token_id=None,
+            max_new_tokens=448,
+        )
+        output = pipe(array, return_timestamps=True, chunk_length_s=30, stride_length_s=[15, 0], device=0)
+        # fmt: off
+        EXPECTED_OUTPUT = {
+            "text": (
+                " A man said to the universe, Sir, I exist. Sweat covered Breon's body, trickling into the tight-wing"
+                " cloth that was the only garment you wore. The cut on his chest still dripping blood. The ache of his"
+                " overstrain dyes. Even the soaring arena around him with thousands of spectators, retrievalidies not"
+                " worth thinking about."
+            ),
+            "chunks": [
+                {"text": " A man said to the universe, Sir, I exist.", "start_offset": 0.0, "end_offset": 5.5},
+                {"text": " Sweat covered Breon's body, trickling into the tight-wing cloth that was the only garment", "start_offset": 5.5, "end_offset": 10.24},
+                {"text": " you wore.", "start_offset": 10.24, "end_offset": 11.74},
+                {"text": " The cut on his chest still dripping blood.", "start_offset": 11.74, "end_offset": 14.88},
+                {"text": " The ache of his overstrain dyes.", "start_offset": 14.88, "end_offset": 17.6},
+                {"text": " Even the soaring arena around him with thousands of spectators, retrievalidies not worth", "start_offset": 17.6, "end_offset": 23.28},
+            ],
+        }
+        # fmt: on
+
+        self.assertDictEqual(output, EXPECTED_OUTPUT)
```

Comment:
`assertEqual` works just as good I think.

Reply:
Also please remove `EXPECTED OUTPUT` temporary variable, it adds little value and creates potential test errors (like misnaming things later on)